prober/icmp.go

```diff
@@ -215,7 +227,25 @@ func ProbeICMP(ctx context.Context, target string, module config.Module, registr
 	durationGaugeVec.WithLabelValues("setup").Add(time.Since(setupStart).Seconds())
 	level.Info(logger).Log("msg", "Writing out packet")
 	rttStart := time.Now()
-	if _, err = socket.WriteTo(wb, dst); err != nil {
+
+	if icmpConn != nil {
+		_, err = icmpConn.WriteTo(wb, dst)
+	} else {
+		header := &ipv4.Header{
```

Comment:
A comment that this is only for DF v4 would be useful, it takes a bit to figure out from the code.

Reply:
// Only for IPv4 raw (needed for setting DontFragment flag)